jupyterhub/user.py

```diff
@@ -235,8 +235,11 @@ def spawn(self, options=None):
         api_token = self.new_api_token()
         db.commit()
 
+        self.server_base_url = base_url
+
         spawner = self.spawner
```

Comment:
A user now needs multiple Spawners, not just one. There needs to be a separate Spawner per server.